### PR TITLE
feat(webhook): add WHATSAPP_WEBHOOK_IGNORE_JIDS to skip chats/JIDs (e.g. groups)

### PR DESCRIPTION
**Pull request by huboperacional**
## Problem

The generic `WHATSAPP_WEBHOOK` forwards everything, including group traffic. There's no native way to drop groups (or specific chats) from the webhook — `WHATSAPP_WEBHOOK_EVENTS` only filters by **event type**. Evolution API has a per-instance `groupsIgnore`; GOWA has no equivalent for the generic webhook.

## Solution

Add `WHATSAPP_WEBHOOK_IGNORE_JIDS`, mirroring the existing `CHATWOOT_IGNORE_JIDS`. It reuses the already-tested `utils.MatchesIgnoredJID`, supports the `@g.us` / `@s.whatsapp.net` / `@lid` address-space wildcards (e.g. `@g.us` drops all group traffic) and exact JIDs, and matches the event's `chat_id` **or** `from`.

It is **orthogonal** to `WHATSAPP_WEBHOOK_EVENTS`: an event is forwarded only if its type is allowed **and** its JID is not ignored. The Chatwoot path keeps its own `CHATWOOT_IGNORE_JIDS`.

## Changes

- `config/settings.go`, `cmd/root.go` — `WHATSAPP_WEBHOOK_IGNORE_JIDS` env + `--webhook-ignore-jids` flag (parsing mirrors `CHATWOOT_IGNORE_JIDS`).
- `infrastructure/whatsapp/webhook_forward.go` — `shouldIgnoreWebhookJID` gate in `forwardPayloadToConfiguredWebhooks`. It descends into the nested inner payload (where `chat_id`/`from` live) and is a defensive no-op when the list is empty, the inner payload is absent, or no JID matches — so events without a JID and the default (no list) keep forwarding unchanged.
- Tests (`webhook_forward_ignore_jids_test.go`) + docs (`.env.example`, `readme.md`, `docs/webhook-payload.md`).

## Notes / edge-cases (for transparency)

A few events carry the group JID elsewhere or omit `chat_id`: `call.offer` uses `group_jid` (not `chat_id`); `message.deleted` only includes `chat_id` when the original message is found locally; `group.participants` / `group.joined` have no `from`. The `@g.us` wildcard still covers ordinary group messages and receipts — the common case for muting groups.

## Testing

`gofmt` (changed files) · `go build ./...` · `go vet ./...` · `go test ./...` · Docker image build — all green.

### CODE REVIEW

**Comment by huboperacional** [comment]
Round-2 pushed in `f360277`. Thanks @coderabbitai and the Codex reviewer.

**Applied**
- **`@lid` wildcard didn't match (Codex):** `shouldIgnoreWebhookJID` now also matches the LID forms `chat_lid` / `from_lid`, not just `chat_id` / `from`. A LID-migrated event keeps the `@lid` JID in the `*_lid` fields while `chat_id` / `from` hold the resolved phone JID, so `@lid` (or an exact `...@lid`) only matches via the `*_lid` fields. Added two tests for it.
- **`@lid` missing from CLI help (CodeRabbit):** the `--webhook-ignore-jids` help text now lists `@lid` alongside `@g.us` / `@s.whatsapp.net`.

**Intentionally not changed (with reasons)**
- **Env overrides the CLI flag (Codex):** this mirrors the existing behavior of every other env var in `cmd/root.go` — e.g. `whatsapp_webhook_events` and `chatwoot_ignore_jids` use the same `if env != "" { config.X = ... }` pattern. Changing CLI-vs-env precedence is a repo-wide concern and out of scope for this PR; happy to follow up separately if you'd like it fixed globally.
- **`@s.whatsapp.net` also suppresses group messages (Codex):** intentional parity with `CHATWOOT_IGNORE_JIDS`, which matches `chat_id` **or** `from` the same way. A group message's `from` is the participant's `@s.whatsapp.net` JID, so that wildcard does affect groups. Documented now, with `@g.us` recommended as the unambiguous way to mute groups.

CI (gofmt + `go build` + `go vet` + `go test` + Docker image) is green.

File: src/infrastructure/whatsapp/webhook_forward.go [hunk 1]
```diff
@@ -116,6 +117,27 @@ func forwardPayloadToConfiguredWebhooks(ctx context.Context, payload map[string]
...
+		return false
+	}
+	data, ok := payload["payload"].(map[string]any)
+	if !ok {
+		return false
+	}
+	chatID, _ := data["chat_id"].(string)
+	from, _ := data["from"].(string)
+	return utils.MatchesIgnoredJID(chatID, config.WhatsappWebhookIgnoreJids) ||
+		utils.MatchesIgnoredJID(from, config.WhatsappWebhookIgnoreJids)
```
[the remark this answers is not in the document]
Comment (aldinokemal) [reply src/infrastructure/whatsapp/webhook_forward.go]: Addressed in f360277: the ignore gate now also matches `chat_lid`/`from_lid`, with tests (`TestWebhookIgnoreJID_LidWildcardMatchesLidFields`, `TestWebhookIgnoreJID_ExactLidMatchesChatLid`).

File: src/cmd/root.go [hunk 2]
```diff
@@ -137,6 +137,16 @@ func initEnvConfig() {
...
 	}
+	if envWebhookIgnoreJids := viper.GetString("whatsapp_webhook_ignore_jids"); envWebhookIgnoreJids != "" {
+		parts := strings.Split(envWebhookIgnoreJids, ",")
+		jids := make([]string, 0, len(parts))
+		for _, p := range parts {
+			if trimmed := strings.TrimSpace(p); trimmed != "" {
+				jids = append(jids, trimmed)
+			}
+		}
+		config.WhatsappWebhookIgnoreJids = jids
```
[the remark this answers is not in the document]
Comment (aldinokemal) [reply src/cmd/root.go:155]: This mirrors how every other setting in `initEnvConfig` behaves (env wins over flag), including the `CHATWOOT_IGNORE_JIDS` pattern this feature copies — keeping it consistent; a repo-wide precedence change belongs in its own PR.

File: src/infrastructure/whatsapp/webhook_forward.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aldinokemal) [reply src/infrastructure/whatsapp/webhook_forward.go]: Matching `from` is intentional (ignore-by-sender); f360277 documented that `@s.whatsapp.net` as a wildcard also suppresses group messages from phone-JID senders, and the group-mute use case is served by `@g.us`.
